homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -0,0 +1,204 @@
+"""Ask tankerkoenig.de for petrol price information."""
+from datetime import timedelta
+from functools import partial
+import logging
+
+import pytankerkoenig
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_RADIUS,
+    CONF_SCAN_INTERVAL,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import CONF_FUEL_TYPES, CONF_STATIONS, DOMAIN, FUEL_TYPES, NAME
+from .sensor import FuelPriceSensor
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_RADIUS = 2
+SCAN_INTERVAL = timedelta(minutes=30)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_API_KEY): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+                vol.Optional(CONF_FUEL_TYPES, default=FUEL_TYPES): vol.All(
+                    cv.ensure_list, [vol.In(FUEL_TYPES)]
+                ),
+                vol.Inclusive(
+                    CONF_LATITUDE,
+                    "coordinates",
+                    "Latitude and longitude must exist together",
+                ): cv.latitude,
+                vol.Inclusive(
+                    CONF_LONGITUDE,
+                    "coordinates",
+                    "Latitude and longitude must exist together",
+                ): cv.longitude,
+                vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS): vol.All(
+                    cv.positive_int, vol.Range(min=1)
+                ),
+                vol.Optional(CONF_STATIONS, default=[]): vol.All(
+                    cv.ensure_list, [cv.string]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set the tankerkoenig component up."""
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    _LOGGER.debug("Setting up tankerkoenig")
```

Comment:
There is no need to add the platform or component name to the log messages. See the [style guidelines](https://developers.home-assistant.io/docs/en/development_guidelines.html#log-messages).